src/semver-intersect.js

```diff
@@ -105,11 +108,29 @@ function intersect (...ranges) {
     return shorthand;
 }
 
+// Prerelease versions must have the exact same major, minor, patch, and prerelease id
+function checkPrereleaseVersions (range, bound) {
+    const parsedRange = parseRange(range);
+    const parsedBound = parseRange(bound);
+
+    if (parsedRange.prerelease && parsedBound.prerelease) {
+        if (
+            (parsedRange.major !== parsedBound.major) ||
+            (parsedRange.minor !== parsedBound.minor) ||
+            (parsedRange.patch !== parsedBound.patch) ||
+            (parsedRange.prerelease[0] !== parsedBound.prerelease[0])
```

Comment:
The prerelease-tag doesn't have to be identical for the range to be vaild, but **a** prerelease tag has to exist:

<img width="441" alt="bildschirmfoto 2018-02-01 um 14 24 13" src="https://user-images.githubusercontent.com/20770029/35680823-b4ce17be-075b-11e8-8d15-dbfc17861927.png">

> If a version has a prerelease tag (for example, 1.2.3-alpha.3) then it will only be allowed to satisfy comparator sets if at least one comparator with **the same [major, minor, patch] tuple** also has **a prerelease tag**.

Reply:
Gotcha, actually it's one more bit above and beyond that - if the `preId` is the same for both then you also have to look at the number associated with it:

```
> s.satisfies('1.0.0-rc4', '^1.0.0-pre3 <2.0.0')
true
> s.satisfies('1.0.0-rc4', '^1.0.0-rc5 <2.0.0')
false
```

Reply:
> The prerelease-tag doesn't have to be identical for the range to be vaild, but a prerelease tag has to exist

So one more weird thing - it's not that the prerelease tag hast to exist, but it *also* takes alphabetical order into account!  rc4 satisfies ^pre but not the other way around.  Crazy